std/string.d

```diff
@@ -350,22 +350,24 @@ alias CaseSensitive = Flag!"caseSensitive";
         If the parameters are not valid UTF, the result will still
         be in the range [-1 .. s.length], but will not be reliable otherwise.
   +/
-ptrdiff_t indexOf(Range)(Range sIn, in dchar c,
+ptrdiff_t indexOf(Range)(auto ref Range s, in dchar c,
         in CaseSensitive cs = CaseSensitive.yes)
-    if ( (isInputRange!Range && isSomeChar!(ElementEncodingType!Range)) ||
-        __traits(compiles, StringTypeOf!Range)
-    )
+    if (!(isInputRange!Range && isSomeChar!(ElementEncodingType!Range))
+        && is(StringTypeOf!Range))
+{
+    return indexOf(cast(StringTypeOf!Range)s, c, cs);
+}
+
+///
```

Comment:
The documentation seems broken at this point. The order is changed and there are some parameter descriptions missing. Needs a "ditto"?